openapi.yaml

```diff
@@ -0,0 +1,149 @@
+        time_period_end: { type: string , description: Trade timestamp, format: "date", example: "1985-04-12T23:15:00.00Z" }
+    Prices:
+      type: array
+      items: {$ref: '#/components/schemas/Price'}
+security:
+  - ApiKeyAuth: [read, write]
+tags: [{name: markets}, {name: orders}, {name: trades}, {name: prices}, {name: settings}]
+externalDocs:
+ description: Bot API Documentation
+ url: https://www.bot-api-docs.geniusyield.co/
+servers:
+  - url: https://localhost:4000/
+  - url: https://bot.api.geniusyield.co/
+paths:
+  /settings:
+    get:
+      tags: [settings]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Settings' }}}}}
+  /tickers:
+    get:
+      tags: [markets]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Tickers' }}}}}
+  /orderbook:
+    parameters:
+     - {in: query, name: ticker_id, required: true, schema: { type: string }}
+     - {in: query, name: depth, required: false, schema: { type: integer }}
+    get:
+      tags: [orders]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Orders' }}}}}
+    post:
+      tags: [orders]
+      responses: {201: {description: Created}}
+    delete:
+      tags: [orders]
+      responses: {202: {description: Deleted}}
+  /historical_trades:
+    parameters:
+     - {in: query, name: ticker_id, required: true, schema: { type: string }}
+     - {in: query, name: type, required: false, schema: { type: string, enum: ["buy", "sell"], example: "buy" }}
+     - {in: query, name: limit, required: false, schema: { type: integer, example: "100" }}
+     - {in: query, name: page, required: false, schema: { type: integer, example: "1" }}
+     - {in: query, name: start_time, required: false, schema: { type: string, format: "date", example: "1996-12-19T16:39:57-08:00" }}
+     - {in: query, name: end_time, required: false, schema: { type: string, format: "date", example: "1996-12-19T17:39:57-08:00" }}
+    get:
+      tags: [trades]
+      responses: { 200: {description: successful operation, content: {application/json: {schema: {$ref: '#/components/schemas/Trades' }}}}}
+  /price_history:
+    parameters:
+     - {in: query, name: ticker_id, required: true, schema: { type: string }}
+     - {in: query, name: resolution, required: false, schema: { type: string, enum: ["1m", "5m", "15m", "30m", "1h", "4h", "1d", "1w", "1mo"], example: "30m" }}
+     - {in: query, name: from, required: false, schema: { type: string, format: "date", example: "2024-12-03" }}
+     - {in: query, name: to, required: false, schema: { type: string, format: "date", example: "2024-12-04" }}
+     - {in: query, name: limit, required: false, schema: { type: integer, example: "100" }}
```

Comment:
Limit doesn't really make much sense without paging.

Paging is not suggested by the Coin Gecko API requirements. It speaks only about limit, but I think adding page AND limit makes much more sense.